larsim/EventGenerator/SingleGen_module.cc

```diff
@@ -447,11 +447,20 @@ namespace evgen {
     , fHistFileName(config().HistogramFile())
     , fPHist(config().PHist())
     , fThetaXzYzHist(config().ThetaXzYzHist())
-    , fEngine(createEngine(0))
+    , fEngine(art::ServiceHandle<rndm::NuRandomService>()->registerAndSeedEngine(createEngine(0),
+                                                                                 "HepJamesRandom",
+                                                                                 "Seed"))
   {
     setup();
     rndm::NuRandomService::seed_t seed;
-    if (config().Seed(seed)) { fEngine.setSeed(seed, 0 /* dummy? */); }
+    if (config().Seed(seed)) {
+      fEngine.setSeed(seed, 0 /* dummy? */);
+      mf::LogInfo("SingleGen") << "Seed set to " << seed << " (from configuration)";
+    }
+    else {
+      seed = fEngine.getSeed();
+      mf::LogInfo("SingleGen") << "Seed set to " << seed;
+    }
```

Comment:
There may be a problem here.
The engine is directly seeded, bypassing `NuRandomService`.
`NuRandomService` does not know that the seed was set manually and should therefore be considered frozen, and it will happily override it if the configured policy does that.

On the other end, using `registerAndSeedEngine()` will implicitly freeze the seed if one is found in the optional parameter specified in the argument.

`NuRandomService` can also print the seeds as used — there are [configuration parameters](https://code-doc.larsoft.org/docs/latest/html/classrndm_1_1SeedMaster.html#details) to make it do that: `services.NuRandomService.endOfJobSummary: true` and/or a non-zero value for `services.NuRandomService.verbose`.
This service should be rewritten in modern _art_ (configuration validation and _art_ tools)... heh.

Reply:
Hi @PetrilloAtWork, sorry if I'm not following the conversation correctly. Do you agree with @knoepfel's proposed solution or are you suggesting an alternative approach? Thanks in advance!

Reply:
We happened to be writing something similar at the same time, unaware of each other.

However I have to check whether there is a difference between telling `NuRandomService` to use seed `0` and not specifying any seed. Chances are that the former would force the seed to be `0`, disabling `NuRandomService` seed management.